Mathlib/GroupTheory/GroupAction/MultipleTransitivity.lean

```diff
@@ -0,0 +1,313 @@
+
+@[to_additive]
+theorem IsPretransitive.of_embedding_congr {n : Type*}
+    (hσ : Function.Surjective σ) (hf : Function.Bijective f) :
+    IsPretransitive G (n ↪ α) ↔ IsPretransitive H (n ↪ β) :=
+  isPretransitive_congr hσ hf.mulActionHom_embedding_isBijective
+
+section Zero
+
+/-- Any action is 0-pretransitive. -/
+@[to_additive]
+theorem is_zero_pretransitive {n : Type*} [IsEmpty n] :
+    IsPretransitive G (n ↪ α) := inferInstance
+
+/-- Any action is 0-pretransitive. -/
+@[to_additive]
+theorem is_zero_pretransitive' :
+    IsMultiplyPretransitive G α 0 := inferInstance
+
+
+end Zero
+
+section One
+
+/-- The equivalence `one ↪ α` with `α`, for `Unique one`. -/
+def _root_.Function.Embedding.oneEmbeddingEquiv {one : Type*} [Unique one] : (one ↪ α) ≃ α where
+  toFun f := f default
+  invFun a := {
+    toFun := fun _ ↦ a
+    inj' x y h := by simp [Unique.uniq inferInstance] }
+  left_inv f := by ext; simp [Unique.uniq]
+  right_inv a := rfl
+
+/-- For `Unique one`, the equivariant map from `one ↪ α` to `α`. -/
+@[to_additive "For `Unique one`, the equivariant map from `one ↪ α` to `α`"]
+def _root_.MulActionHom.oneEmbeddingMap {one : Type*} [Unique one] :
+    (one ↪ α) →[G] α := {
+  oneEmbeddingEquiv with
+  map_smul' _ _ := rfl }
+
+@[to_additive]
+theorem _root_.MulActionHom.oneEmbeddingMap_bijective {one : Type*} [Unique one] :
+    Function.Bijective (oneEmbeddingMap (one := one) (G := G) (α := α)) :=
+  oneEmbeddingEquiv.bijective
+
+/-- An action is 1-pretransitive iff it is pretransitive. -/
+@[to_additive]
+theorem oneEmbedding_isPretransitive_iff {one : Type*} [Unique one] :
+    IsPretransitive G (one ↪ α) ↔ IsPretransitive G α :=
+  isPretransitive_congr Function.surjective_id oneEmbeddingMap_bijective
+
+/-- An action is 1-pretransitive iff it is pretransitive. -/
+@[to_additive]
+theorem is_one_pretransitive_iff :
+    IsMultiplyPretransitive G α 1 ↔ IsPretransitive G α :=
+  oneEmbedding_isPretransitive_iff
+
+end One
+
+section Two
+
+/-- Two distinct elements of `α` give an embedding `Fin 2 ↪ α`. -/
+def _root_.Function.Embedding.embFinTwo {a b: α} (h : a ≠ b) : Fin 2 ↪ α where
+  toFun := ![a, b]
+  inj' i j hij := by
+    by_cases hi : i = 0
+    · by_cases hj : j = 0
+      · rw [hi, hj]
+      · simp [hi, eq_one_of_ne_zero j hj] at hij;
+        exact False.elim (h hij)
+    · rw [eq_one_of_ne_zero i hi] at hij ⊢
+      by_cases hj : j = 0
+      · simp [hj] at hij; exact False.elim (h hij.symm)
+      · rw [eq_one_of_ne_zero j hj]
+
+theorem _root_.Function.Embedding.embFinTwo_apply_zero
+    {a b : α} (h : a ≠ b) : embFinTwo h 0 = a := rfl
+
+theorem _root_.Function.Embedding.embFinTwo_apply_one
+    {a b : α} (h : a ≠ b) : embFinTwo h 1 = b := rfl
+
+/-- An action is 2-pretransitive iff
+it can move any two distinct elements to any two distinct elements. -/
+@[to_additive "An additive action is 2-pretransitive iff
+it can move any two distinct elements to any two distinct elements."]
```

Comment:
```suggestion
/-- An action is `2`-pretransitive iff
it can move any two distinct elements to any two distinct elements. -/
@[to_additive "An additive action is `2`-pretransitive iff
it can move any two distinct elements to any two distinct elements."]
```